ua/encode.go

```diff
@@ -79,25 +79,25 @@ func encode(val reflect.Value, name string) ([]byte, error) {
 			if val.IsNil() {
 				return nil, nil
 			}
-			return logIntercept(encode(val.Elem(), name))
+			return dump(encode(val.Elem(), name))
 		case reflect.Struct:
-			return logIntercept(writeStruct(val, name))
+			return dump(writeStruct(val, name))
 		case reflect.Slice:
-			return logIntercept(writeSlice(val, name))
+			return dump(writeSlice(val, name))
 		case reflect.Array:
-			return logIntercept(writeArray(val, name))
+			return dump(writeArray(val, name))
 		default:
 			return nil, errors.Errorf("unsupported type: %s", val.Type())
 		}
 	}
-	return logIntercept(buf.Bytes(), buf.Error())
+	return dump(buf.Bytes(), buf.Error())
 }
 
-func logIntercept(bytes []byte, err error) ([]byte, error) {
+func dump(b []byte, err error) ([]byte, error) {
 	if debugCodec {
-		fmt.Printf("Wrote:\n%s\n", hex.Dump(bytes))
+		fmt.Printf("Wrote:\n%s\n", hex.Dump(b))
```

Comment:
I've played a bit with the output and think this is better:

```go
fmt.Printf("%s---\n", hex.Dump(b))
```

```
encode: *ua.CloseSecureChannelRequest.RequestHeader.TimeoutHint has type uint32 and is a uint32
00000000  10 27 00 00                                       |.'..|
---
encode: *ua.CloseSecureChannelRequest.RequestHeader.AdditionalHeader has type *ua.ExtensionObject and is a ptr
00000000  00 00 00                                          |...|
---
00000000  00 00 a2 49 62 c7 46 d7  d9 01 07 00 00 00 00 00  |...Ib.F.........|
00000010  00 00 ff ff ff ff 10 27  00 00 00 00 00           |.......'.....|
---
00000000  00 00 a2 49 62 c7 46 d7  d9 01 07 00 00 00 00 00  |...Ib.F.........|
00000010  00 00 ff ff ff ff 10 27  00 00 00 00 00           |.......'.....|
---
00000000  00 00 a2 49 62 c7 46 d7  d9 01 07 00 00 00 00 00  |...Ib.F.........|
00000010  00 00 ff ff ff ff 10 27  00 00 00 00 00           |.......'.....|
---
00000000  00 00 a2 49 62 c7 46 d7  d9 01 07 00 00 00 00 00  |...Ib.F.........|
00000010  00 00 ff ff ff ff 10 27  00 00 00 00 00           |.......'.....|
---
```

instead of 

```
encode: *ua.CloseSessionRequest.RequestHeader.TimeoutHint has type uint32 and is a uint32
Wrote:
00000000  10 27 00 00                                       |.'..|

encode: *ua.CloseSessionRequest.RequestHeader.AdditionalHeader has type *ua.ExtensionObject and is a ptr
Wrote:
00000000  00 00 00                                          |...|

Wrote:
00000000  02 00 00 ee 03 00 00 ba  22 ab a6 46 d7 d9 01 06  |........"..F....|
00000010  00 00 00 00 00 00 00 ff  ff ff ff 10 27 00 00 00  |............'...|
00000020  00 00                                             |..|

Wrote:
00000000  02 00 00 ee 03 00 00 ba  22 ab a6 46 d7 d9 01 06  |........"..F....|
00000010  00 00 00 00 00 00 00 ff  ff ff ff 10 27 00 00 00  |............'...|
00000020  00 00                                             |..|
```

Reply:
sure, I pushed this variant